packages/block-library/src/post-author/edit.js

```diff
@@ -70,28 +70,30 @@ function PostAuthorEdit( {
 		<>
 			<InspectorControls>
 				<PanelBody title={ __( 'Settings' ) }>
-					{ ! isDescendentOfQueryLoop && !! authors?.length && (
-						<SelectControl
-							label={ __( 'Author' ) }
-							value={ authorId }
-							options={ authors.map( ( { id, name } ) => {
-								return {
-									value: id,
-									label: name,
-								};
-							} ) }
-							onChange={ ( nextAuthorId ) => {
-								editEntityRecord(
-									'postType',
-									postType,
-									postId,
-									{
-										author: nextAuthorId,
-									}
-								);
-							} }
-						/>
-					) }
+					{ !! postId &&
+						! isDescendentOfQueryLoop &&
+						!! authors?.length && (
```

Comment:
Do both `!! postId` and `! isDescendentOfQueryLoop` need to be checked?

Reply:
I have removed the _! isDescendentOfQueryLoop_ and run some tests and everything seems to be fine. But are we sure that _ isDescendentOfQueryLoop_ will be false each time when _ postId_ will be defined? is there a way to confirm that? Thanks.

Reply:
Does postId get passed via context to author blocks in query loops when the query loop is in a post? if it does then we need to keep `! isDescendentOfQueryLoop` otherwise no.